src/senselab/utils/data_structures/dataset.py

```diff
@@ -0,0 +1,90 @@
+"""Data structures relevant for managing datasets."""
+
+import uuid
+from typing import Any, Dict, List
+
+from pydantic import BaseModel, Field, field_validator
+
+
+class Participant(BaseModel):
+    """Data structure for a participant in a dataset."""
+
+    id: str = Field(default_factory=lambda: str(uuid.uuid4()))
+    metadata: Dict = Field(default={})
+
+    @field_validator("id", mode="before")
+    def set_id(cls, v: str) -> str:
+        """Set the unique id of the participant."""
+        return v or str(uuid.uuid4())
+
+    def __eq__(self, other: object) -> bool:
+        """Overloads the default BaseModel equality to correctly check that ids are equivalent."""
+        if isinstance(other, Participant):
+            return self.id == other.id
+        return False
+
+
+class Session(BaseModel):
+    """Data structure for a session in a dataset."""
+
+    id: str = Field(default_factory=lambda: str(uuid.uuid4()))
+    metadata: Dict = Field(default={})
+
+    @field_validator("id", mode="before")
+    def set_id(cls, v: str) -> str:
+        """Set the unique id of the session."""
+        return v or str(uuid.uuid4())
+
+    def __eq__(self, other: object) -> bool:
+        """Overloads the default BaseModel equality to correctly check that ids are equivalent."""
+        if isinstance(other, Session):
+            return self.id == other.id
+        return False
+
+
+class SenselabDataset(BaseModel):
```

Comment:
seems like we made complimentary datasets (mine handles Audio/Video, yours has some base functions for the participants and sessions).

Reply:
sounds amazing. If you want, we can make a coding session together tomorrow after lunch and marge all what we have